nuxeo-runtime/nuxeo-stream/src/main/java/org/nuxeo/lib/stream/log/kafka/KafkaUtils.java

```diff
@@ -237,16 +247,18 @@ protected AdminClient getAdminClient() {
     public List<String> listAllConsumers() {
         long now = System.currentTimeMillis();
         if (allConsumers == null || (now - allConsumersTime) > ALL_CONSUMERS_CACHE_TIMEOUT_MS) {
-            allConsumers = new ArrayList<>();
-            // this returns only consumer group that use the subscribe API (known by coordinator)
-            scala.collection.immutable.List<GroupOverview> groups = getAdminClient().listAllConsumerGroupsFlattened();
-            Iterator<GroupOverview> iterator = groups.iterator();
-            GroupOverview group;
-            while (iterator.hasNext()) {
-                group = iterator.next();
-                if (group != null && !allConsumers.contains(group.groupId())) {
-                    allConsumers.add(group.groupId());
-                }
+            try {
+                allConsumers = getAdminClient().listConsumerGroups()
+                                               .all()
+                                               .get()
+                                               .stream()
+                                               .map(group -> group.groupId())
+                                               .collect(Collectors.toList());
+            } catch (InterruptedException e) {
+                Thread.currentThread().interrupt();
+                throw new StreamRuntimeException(e);
+            } catch (ExecutionException e) {
+                throw new StreamRuntimeException(e);
             }
             if (!allConsumers.isEmpty()) {
                 allConsumersTime = now;
```

Comment:
Hm `allConsumersTime` must really be `volatile`, and the update to the `allConsumers` fields seems weird as it's always done even if `allConsumersTime` isn't changed. I'd use a local variable first, and assign to `allConsumers` only if you change the time (and `allConsumers` should be `volatile` too then).

Reply:
indeed the code is not thread-safe but the cache is working as expected, I will fix this but this is unrelated to the change.